### PR TITLE
Bdio 2

### PR DESCRIPTION
**Pull request by JakeMathews**
Adds support for BDIO 2 to blackduck-common

### CODE REVIEW

**Comment by JakeMathews** [comment]
Will be adding tests soon.

File: src/main/java/com/synopsys/integration/blackduck/service/BlackDuckService.java [hunk 1]
```diff
@@ -58,6 +58,7 @@
     public static final BlackDuckPath BOMIMPORT_PATH = new BlackDuckPath("/api/bom-import");
     public static final BlackDuckPath SCANSUMMARIES_PATH = new BlackDuckPath("/api/scan-summaries");
     public static final BlackDuckPath UPLOADS_PATH = new BlackDuckPath("/api/uploads");
+    public static final BlackDuckPath SCAN_DATA_PATH = new BlackDuckPath("/api/scan/data");
```
Comment (ketul-shukla) [inline src/main/java/com/synopsys/integration/blackduck/service/BlackDuckService.java]: This api would need to have / at the end. 
eg: /api/scan/data/

Comment (ekerwin) [reply src/main/java/com/synopsys/integration/blackduck/service/BlackDuckService.java]: I don't believe that is accurate, this is the correct way for the BlackDuckPath to be constructed.

Comment (JakeMathews) [reply src/main/java/com/synopsys/integration/blackduck/service/BlackDuckService.java]: So I just tested and without the slash I get a 404 status code and with the slash I get a 201 status code. Not sure why this endpoint is different.

Comment (ekerwin) [reply src/main/java/com/synopsys/integration/blackduck/service/BlackDuckService.java]: That sounds like a bug somewhere.

**Comment by ekerwin** [comment]
I don't believe this code will be used for a Polaris upload, but if there are requirements to support that end point correctly, they should likely be spelled out in the doc for the api.

File: src/main/java/com/synopsys/integration/blackduck/service/BlackDuckService.java (hunk 1 above)
Comment (ketul-shukla) [reply src/main/java/com/synopsys/integration/blackduck/service/BlackDuckService.java]: The API inside Blackduck is /scan/data/ that's the reason why I mentioned that.

**Pull request merged**